### PR TITLE
Db interface cachedir

### PR DESCRIPTION
**Pull request by chengxuz**
I fix both the DB interface cachedir bug and loss function bug for train_alexnet in newest version of tensorflow. See issue #35 and issue #36

### CODE REVIEW

**Comment by qbilius** [comment]
I slightly modified your solution for `cross_entropy`.
